internal/datanode/collection_replica.go

```diff
@@ -338,3 +348,30 @@ func (replica *CollectionSegmentReplica) hasCollection(collectionID UniqueID) bo
 	_, ok := replica.collections[collectionID]
 	return ok
 }
+
+// setStartPositions set segment `Start Position` - means the `startPositions` from the MsgPack when segment is first found
+func (replica *CollectionSegmentReplica) setStartPositions(segID UniqueID, startPositions []*internalpb.MsgPosition) error {
+	replica.posMu.Lock()
+	defer replica.posMu.Unlock()
+	replica.startPositions[segID] = startPositions
+	return nil
+}
+
+// setEndPositions set segment `End Position` - means the `startPositions` from the MsgPack when segment need to be flushed
```

Comment:
I'm not sure about this. Maybe need @xige-16 's help.

Reply:
According to @neza2017, segment End Position shall be the endPositions from MsgPack.